### PR TITLE
Allow services to report config files as invalid

### PR DESCRIPTION
**Pull request by teoljungberg**
Either by executing the service, `ReportInvalidConfig`. Or by enqueueing
the corresponding job, `ReportInvalidConfigJob`.

https://trello.com/c/42U5K0Bs

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -173,6 +173,14 @@ Once linting is complete, resulting violations should be posted to the outbound
 * `patch` - The patch content from GitHub for the file being reviewed. This is
   provided by the inbound queue.
 
+If the given `config` is invalid, queue invalid file the outbound
```
Comment (thorncp) [inline CONTRIBUTING.md]: What do you think about keeping the verbiage consistent with the sentence above?

Maybe

> If the given `config` is invalid, the invalid file should be posted to the outbound `InvalidConfigFileJob` queue:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -173,6 +173,14 @@ Once linting is complete, resulting violations should be posted to the outbound
...
+`InvalidConfigFileJob` queue:
+
+* `commit_sha` - The git commit SHA of the code snippet.
+* `filename` - The name of the source file for the code snippet.
+* `pull_request_number` - The GitHub pull request number. This is provided by the
```
Comment (thorncp) [inline CONTRIBUTING.md]: All three of these bullet points are provided by the inbound queue.

File: app/jobs/invalid_config_file_job.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class InvalidConfigFileJob
+  @queue = :high
+
+  def self.perform(attributes)
+    InvalidConfigFile.run(
```
Comment (thorncp) [inline app/jobs/invalid_config_file_job.rb]: This reads a little odd to me. What does it mean for a file to run? I don't have a suggestions :trollface:

**Comment by thorncp** [comment]
A few questions/suggestions, but looks good overall.

File: app/jobs/invalid_config_file_job.rb (hunk 3 above)
Comment (teoljungberg) [inline app/jobs/invalid_config_file_job.rb]: Yeah, good point.
I wanted to have the pattern where the job-class only enqueued a service. Would a better name than `InvalidConfigFile` make it clearer? I don't have any suggestions

Comment (teoljungberg) [inline app/jobs/invalid_config_file_job.rb]: `InvalidateConfig` maybe?

File: CONTRIBUTING.md (hunk 1 above)
Comment (teoljungberg) [inline CONTRIBUTING.md]: Good idea.

File: CONTRIBUTING.md (hunk 2 above)
Comment (teoljungberg) [inline CONTRIBUTING.md]: Good call, I added the explanation to the docs

File: app/jobs/invalid_config_file_job.rb (hunk 3 above)
Comment (thorncp) [inline app/jobs/invalid_config_file_job.rb]: Hmm, _invalidate_ makes it sound like Hound is causing the error. Maybe a verb that indicates we're informing someone about the error? `ReportInvalidConfig`? `PostInvalidConfig`?

Comment (teoljungberg) [inline app/jobs/invalid_config_file_job.rb]: `ReportInvalidConfig` sounds like a good name, thanks!

**Pull request merged**